MusicNotationKit/MusicNotationKit/Measure.swift

```diff
@@ -53,50 +55,327 @@ public struct Measure: ImmutableMeasure, Equatable {
+
+        while expectedCount > 0 && notes[setIndex].count > index{
+            tupletNotes.append(notes[setIndex][index])
+            expectedCount -= notes[setIndex][index].noteCount
+            index += 1
+        }
+
+        let newTuplet = try Tuplet(count, baseNoteDuration, inSpaceOf: baseCount, notes: tupletNotes)
+        try newMeasure.removeNotesInRange(noteRange, inSet: setIndex)
+        try newMeasure.insert(newTuplet, at: noteRange.lowerBound, inSet: setIndex)
+        self = newMeasure
+    }
+
+    public mutating func breakdownTuplet(at index: Int, inSet setIndex: Int = 0) throws {
+        var newMeasure = self
+        let collectionIndex = try newMeasure.noteCollectionIndex(fromNoteIndex: index, inSet: setIndex)
+        guard let tupletIndex = collectionIndex.tupletIndex, tupletIndex == 0 else {
+            throw MeasureError.invalidTupletIndex
+        }
+        guard let tuplet = newMeasure.notes[setIndex][collectionIndex.noteIndex] as? Tuplet else {
+            assertionFailure("note collection should be tuplet, but cast failed")
+            throw MeasureError.internalError
+        }
+        newMeasure.notes[setIndex].remove(at: collectionIndex.noteIndex)
+
+        for note in tuplet.notes.reversed() {
+            newMeasure.notes[setIndex].insert(note, at: collectionIndex.noteIndex)
+        }
+        self = newMeasure
+    }
+
+    internal mutating func insert(_ noteCollection: NoteCollection, at index: Int, inSet setIndex: Int = 0, skipTies skipTieConfig: Bool) throws {
+        if index == 0 && notes[setIndex].count == 0 {
+            append(noteCollection, inSet: setIndex)
+            return
+        }
+
+        // If the index points to the end of the note collection, then
+        // append entry.
+        guard index != noteCount[setIndex] else {
+            append(noteCollection, inSet: setIndex)
+            return
+        }
+
+        var newMeasure = self
+        let collectionIndex = try newMeasure.noteCollectionIndex(fromNoteIndex: index, inSet: setIndex)
+
+        guard collectionIndex.tupletIndex == nil else {
+            throw MeasureError.invalidTupletIndex
+        }
+
+        if !skipTieConfig {
+            try newMeasure.prepTiesForInsertion(at: index, inSet: setIndex)
+        }
+        newMeasure.notes[setIndex].insert(noteCollection, at: collectionIndex.noteIndex)
+        self = newMeasure
+    }
+
+    internal mutating func insert(_ noteCollections: [NoteCollection], at index: Int, inSet setIndex: Int = 0, skipTies skipTieConfig: Bool) throws {
+        var newMeasure = self
+        for noteCollection in noteCollections.reversed() {
+            try newMeasure.insert(noteCollection, at: index, inSet: setIndex, skipTies: skipTieConfig)
+        }
+        self = newMeasure
+    }
+
+    internal mutating func replaceNote(at collectionIndex: NoteCollectionIndex, with noteCollection: NoteCollection, inSet setIndex: Int) throws {
```

Comment:
Just take this and make it take an array for the new `noteCollection`. Should function the same. This resolves comment below.